src/clj/fluree/db/virtual_graph/proto.cljc

```diff
@@ -0,0 +1,7 @@
+(ns fluree.db.virtual-graph.proto)
+
+(defprotocol UpdatableVirtualGraph
+  (upsert [this source-db new-flakes remove-flakes] "Updates the virtual graph with the provided flakes. Returns async chan with new updated VirtualGraph or exception.")
+  (initialize [this source-db] "Initialize a new virtual graph based on the provided db - returns promise chan of eventual result")
+  (serialize [this] "Returns a JSON serializable representation of the virtual graph (does not serialize to JSON)")
+  (deserialize [this source-db data] "Reifies the virtual graph from the provided data structure"))
```

Comment:
I think we might be able to adapt the existing [`fluree.db.serde.protocol/StorageSerializer` protocol](https://github.com/fluree/db/blob/43013bc5020b3b180330950f02bae404b83f8cdf/src/clj/fluree/db/serde/protocol.cljc#L6-L14) to work with virtual graphs.